minCostFlow.py

```diff
@@ -41,10 +45,40 @@ def construct_digraph(edges_file, cap):
             if not node2 in idDict:
                 idDict[node2] = curID
                 curID += 1
-            #Google's solver can only handle int weights, so round to the 100th
-            w = int((1-(float(tokens[2])))*100)
-            G.add_arc_with_capacity_and_unit_cost(idDict[node1],idDict[node2], default_capacity, int(w))
-            G.add_arc_with_capacity_and_unit_cost(idDict[node2],idDict[node1], default_capacity, int(w))
+            # Google's solver can only handle int weights, so round to the 100th 
+            w = int((1-(float(tokens[2])))*100) # lower the weight from token[2], higher the cost
+            d = tokens[3]
+            edge = (node1, node2)
+            sorted_edge = tuple(sorted(edge))
+            
+            if d == "D":
+                if edge in directed_dict:
+                    if w < directed_dict[edge]:
+                        directed_dict[edge] = w
+                elif sorted_edge in undirected_dict:
+                    del undirected_dict[sorted_edge]
+                    directed_dict[edge] = w
+                else: # edge not in directed_dict 
+                    directed_dict[edge] = w
+
+            elif d == "U":
+                if edge not in directed_dict and sorted_edge not in directed_dict and sorted_edge not in undirected_dict:
```

Comment:
What if edge and sorted_edge are the same order but the reverse of edge is in the directed_dict?

Reply:
Yeah this case can defiantly happen